ground_truth_labeling_jobs/ground_truth_conversion_scripts/GT_semantic_segmentation_to_COCO.ipynb

```diff
@@ -0,0 +1,405 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Semantic Segmentation Job output conversion to COCO format\n",
+    "This notebook walks through converting the output from SageMaker Ground Truth Semantic segmentation task into Common Objects in Context (COCO) format. The output manifest of the semantic segmentation task contains a reference to a PNG file masks for the objects that has been annotated and saved in an Amazon S3 bucket. In this notebook, I will download a sample manifest file, convert the mask into a pixelated format as in the COCO format."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install Pre-requisites"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install opencv-python pycocotools"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get the output.manifest file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import os\n",
+    "\n",
+    "JOBNAME = '<LABELING JOB NAME>' #Replace it with the labeling job name\n",
+    "REGION = '<REGION>' #Replace it with the job region\n",
+    "client = boto3.client('sagemaker',region_name=REGION)\n",
+    "response = client.describe_labeling_job(LabelingJobName= JOBNAME)\n",
+    "file = response['LabelingJobOutput']['OutputDatasetS3Uri']\n",
+    "output_manifest = os.path.basename(file)\n",
+    "!aws s3 cp $file ./"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Read the manifest file line by line and create \"Images\" key"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "data_objs  = []\n",
+    "input_files = []\n",
+    "images = []\n",
+    "file_name = os.path.join(\"./\", output_manifest)\n",
+    "\n",
+    "#Assuming all images are reshaped to have the same dimensions\n",
+    "height = 3956\n",
+    "width = 5280\n",
+    "\n",
+    "#Get data objects from the output manifest\n",
+    "with open(file_name) as out_manifest:\n",
+    "    for line in out_manifest:\n",
+    "        data_objs.append(json.loads(line))\n",
+    "\n",
+    "#Get the input images filenames\n",
+    "for line in data_objs:\n",
+    "        input_files.append(os.path.basename(line['source-ref']))\n",
```

Comment:
Similar to the issue below: You're assuming that all the annotated images have distinct file basenames. If you have something like `'s3://some-bucket/foo/0.png'` and `'s3://some-bucket/bar/0.png'` in the same output manifest, both are going to be put into COCO_file.json with the same `'file_name'` attribute, `'0.png'`, and you won't be able to tell which annotation goes with which image.

Reply:
resolved